src/System.Management.Automation/FormatAndOutput/common/PSStyle.cs

```diff
@@ -0,0 +1,414 @@
+        }
+
+        /// <summary>
+        /// Gets or sets the rendering mode for output.
+        /// </summary>
+        public OutputRendering OutputRendering
+        {
+            get
+            {
+                return Utils.OutputRenderingSetting;
+            }
+
+            set
+            {
+                Utils.OutputRenderingSetting = value;
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets value to turn off all attributes.
+        /// </summary>
+        public string Reset { get; set; } = "\x1b[0m";
+
+        /// <summary>
+        /// Gets or sets value to turn off blink.
+        /// </summary>
+        public string BlinkOff { get; set; } = "\x1b[5m";
+
+        /// <summary>
+        /// Gets or sets value to turn on blink.
+        /// </summary>
+        public string Blink { get; set; } = "\x1b[25m";
+
+        /// <summary>
+        /// Gets or sets value to turn off bold.
+        /// </summary>
+        public string BoldOff { get; set; } = "\x1b[22m";
+
+        /// <summary>
+        /// Gets or sets value to turn on blink.
+        /// </summary>
+        public string Bold { get; set; } = "\x1b[1m";
+
+        /// <summary>
+        /// Gets or sets value to turn on hidden.
+        /// </summary>
+        public string Hidden { get; set; } = "\x1b[8m";
+
+        /// <summary>
+        /// Gets or sets value to turn off hidden.
+        /// </summary>
+        public string HiddenOff { get; set; } = "\x1b[28m";
+
+        /// <summary>
+        /// Gets or sets value to turn on reverse.
+        /// </summary>
+        public string Reverse { get; set; } = "\x1b[7m";
+
+        /// <summary>
+        /// Gets or sets value to turn off reverse.
+        /// </summary>
+        public string ReverseOff { get; set; } = "\x1b[27m";
+
+        /// <summary>
+        /// Gets or sets value to turn off standout.
+        /// </summary>
+        public string ItalicOff { get; set; } = "\x1b[23m";
+
+        /// <summary>
+        /// Gets or sets value to turn on standout.
+        /// </summary>
+        public string Italic { get; set; } = "\x1b[3m";
+
+        /// <summary>
+        /// Gets or sets value to turn off underlined.
+        /// </summary>
+        public string UnderlineOff { get; set; } = "\x1b[24m";
+
+        /// <summary>
+        /// Gets or sets value to turn on underlined.
+        /// </summary>
+        public string Underline { get; set; } = "\x1b[4m";
+
+        /// <summary>
+        /// Gets or sets the formatting rendering settings.
+        /// </summary>
+        public FormattingData Formatting { get; set; }
+
+        /// <summary>
+        /// Gets or sets foreground colors.
+        /// </summary>
+        public ForegroundColor Foreground { get; set; }
+
+        /// <summary>
+        /// Gets or sets background colors.
+        /// </summary>
+        public BackgroundColor Background { get; set; }
```

Comment:
For `Foreground` and `Background` properties, now that both types only contain read-only properties, it seems the setter here is not needed any more.

Reply:
yup